webapp/channels/src/components/user_settings/general/user_settings_general.tsx

```diff
@@ -702,13 +709,22 @@ export class UserSettingsGeneralTab extends PureComponent<Props, State> {
                                 />
                             </label>
                             <div className='col-sm-7'>
-                                <input
+                                <Input
                                     id='currentPassword'
-                                    className='form-control'
+                                    name='currentPassword'
                                     type='password'
                                     onChange={this.updateCurrentPassword}
                                     value={this.state.currentPassword}
                                     aria-label={formatMessage({id: 'user.settings.general.currentPassword', defaultMessage: 'Current Password'})}
+                                    validate={() => {
```

Comment:
As an extra thing that's certainly not necessary, if we make the change where this function receives the value, we could have standard validators that would be reusable. They'd probably have to be some sort of factory to make the error text more specific, but 

```jsx
function validateNonEmpty(errorMessage: ReactNode) {
    return (value: string) => {
        if (value) {
            return errorMessage;
        } else {
            return undefined;
        }
    };
}

const nonEmptyPassword = validateNonEmpty(<FormattedMessage {...holders.emptyPassword}/>);

function UserSettingsGeneral() {
    return (
        <Input
            name='somethingNonEmpty'
            validate={validateNonEmpty}
        />
    );
}
```